contractcourt/channel_arbitrator_test.go

```diff
@@ -2571,8 +2591,7 @@ func TestChannelArbitratorAnchors(t *testing.T) {
 		// incoming HTLC (toRemoteHTLCs) has a lower CLTV.
 		Htlcs: []channeldb.HTLC{htlc, htlcWithPreimage},
 	}
-	err = chanArb.notifyContractUpdate(newUpdate)
-	require.NoError(t, err)
```

Comment:
Is it feasible to add a _new_ test that either reproduces the deadlock, or demonstrates an inconsistency that could arise due to the existing logic?

Reply:
I have a local itest but IMO doesn't really belong in this repo, I did verify that the deadlock no longer occurs though. I don't think a unit test could reproduce it